### ISSUE TITLE
error on freemyip.com with ssl

### ISSUE THREAD

**Issue by queeup** [troglobit/inadyn#325]
I am using latest docker container. There is a problem with freemyip.com and ssl. 
```console
$ docker image inspect --format '{{ .RepoDigests }}' troglobit/inadyn:latest 
[troglobit/inadyn@sha256:b341794acbf2ff8207e246f7b12eeb0b87242b38e552b8bf69d75f678caff04d]
```

#### Debug log with ssl:
```logs
inadyn[1]: In-a-dyn version 2.7 -- Dynamic DNS update client.
inadyn[1]: Guessing DDNS plugin 'default@freemyip.com' from 'freemyip'
inadyn[1]: Startup delay: 15 sec ...
inadyn[1]: Resolving hostname MY_DOMAIN_NAME.freemyip.com => IP# MY_IP_ADDRESS
inadyn[1]: Base64 encoded string: SECRET
inadyn[1]: Get address for default@freemyip.com
inadyn[1]: Checking for IP# change, initiating HTTPS ...
inadyn[1]: Checking for IP# change, connecting to ipv4.wtfismyip.com([69.195.159.158]:443)
inadyn[1]: Certificate OK
inadyn[1]: SSL server cert subject: CN=wtfismyip.com
inadyn[1]: SSL server cert issuer: C=US,O=Let's Encrypt,CN=Let's Encrypt Authority X3
inadyn[1]: Querying DDNS checkip server for my public IP#: GET /text HTTP/1.0
Host: ipv4.wtfismyip.com
User-Agent: inadyn/2.7 https://github.com/troglobit/inadyn/issues

inadyn[1]: Successfully sent HTTPS request!
inadyn[1]: Successfully received HTTPS response (181/8191 bytes)!
inadyn[1]: Server response: HTTP/1.0 200 OK
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
Content-Type: text/plain
Date: Wed, 18 Nov 2020 22:40:38 GMT
Content-Length: 12

MY_IP_ADDRESS
inadyn[1]: Checked my IP, return code 0: OK
inadyn[1]: IP server response:
inadyn[1]: HTTP/1.0 200 OK
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
Content-Type: text/plain
Date: Wed, 18 Nov 2020 22:40:38 GMT
Content-Length: 12

MY_IP_ADDRESS
inadyn[1]: Checking IPv4 address MY_IP_ADDRESS ...
inadyn[1]: IPv4 address MY_IP_ADDRESS is valid.
inadyn[1]: No IP# change detected for default@freemyip.com, still at MY_IP_ADDRESS
inadyn[1]: Update forced for alias MY_DOMAIN_NAME.freemyip.com, new IP# MY_IP_ADDRESS
inadyn[1]: Sending IP# update to DDNS server, initiating HTTPS ...
inadyn[1]: Sending IP# update to DDNS server, connecting to freemyip.com([35.166.142.92]:443)
inadyn[1]: Certificate OK
inadyn[1]: SSL server cert subject: CN=freemyip.com
inadyn[1]: SSL server cert issuer: C=US,O=Let's Encrypt,CN=Let's Encrypt Authority X3
inadyn[1]: Sending alias table update to DDNS server: GET /update?token=MY_TOKEN&domain=MY_DOMAIN_NAME.freemyip.com HTTP/1.0
Host: freemyip.com
User-Agent: inadyn/2.7 https://github.com/troglobit/inadyn/issues

inadyn[1]: Successfully sent HTTPS request!
inadyn[1]: Successfully received HTTPS response (145/8191 bytes)!
inadyn[1]: DDNS server response: HTTP/1.0 200 OK
Content-Type: text/plain
Content-Length: 3
Connection: close
Date: Wed, 18 Nov 2020 22:40:49 GMT
Server: lighttpd/1.4.35

inadyn[1]: Fatal error in DDNS server response:
inadyn[1]: [200 OK] 
inadyn[1]: Error response from DDNS server, exiting!
inadyn[1]: Error code 48: DDNS server response not OK
```

#### Debug log without ssl:
```logs
inadyn[1]: In-a-dyn version 2.7 -- Dynamic DNS update client.
inadyn[1]: Guessing DDNS plugin 'default@freemyip.com' from 'freemyip'
inadyn[1]: Startup delay: 15 sec ...
inadyn[1]: STOP command received, exiting.
inadyn[1]: In-a-dyn version 2.7 -- Dynamic DNS update client.
inadyn[1]: Guessing DDNS plugin 'default@freemyip.com' from 'freemyip'
inadyn[1]: Startup delay: 15 sec ...
inadyn[1]: Resolving hostname MY_DOMAIN_NAME.freemyip.com => IP# MY_IP_ADDRESS
inadyn[1]: Base64 encoded string: SECRET
inadyn[1]: Get address for default@freemyip.com
inadyn[1]: Checking for IP# change, connecting to ipv4.wtfismyip.com([69.195.159.158]:80)
inadyn[1]: Querying DDNS checkip server for my public IP#: GET /text HTTP/1.0
Host: ipv4.wtfismyip.com
User-Agent: inadyn/2.7 https://github.com/troglobit/inadyn/issues

inadyn[1]: Server response: HTTP/1.0 200 OK
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
Content-Type: text/plain
Date: Wed, 18 Nov 2020 22:42:55 GMT
Content-Length: 12

MY_IP_ADDRESS
inadyn[1]: Checked my IP, return code 0: OK
inadyn[1]: IP server response:
inadyn[1]: HTTP/1.0 200 OK
Access-Control-Allow-Methods: GET
Access-Control-Allow-Origin: *
Content-Type: text/plain
Date: Wed, 18 Nov 2020 22:42:55 GMT
Content-Length: 12

MY_IP_ADDRESS
inadyn[1]: Checking IPv4 address MY_IP_ADDRESS ...
inadyn[1]: IPv4 address MY_IP_ADDRESS is valid.
inadyn[1]: No IP# change detected for default@freemyip.com, still at MY_IP_ADDRESS
inadyn[1]: Update forced for alias MY_DOMAIN_NAME.freemyip.com, new IP# MY_IP_ADDRESS
inadyn[1]: Sending IP# update to DDNS server, connecting to freemyip.com([35.166.142.92]:80)
inadyn[1]: Sending alias table update to DDNS server: GET /update?token=MY_TOKEN&domain=MY_DOMAIN_NAME.freemyip.com HTTP/1.0
Host: freemyip.com
User-Agent: inadyn/2.7 https://github.com/troglobit/inadyn/issues

inadyn[1]: DDNS server response: HTTP/1.0 200 OK
Content-Type: text/plain
Content-Length: 3
Connection: close
Date: Wed, 18 Nov 2020 22:43:06 GMT
Server: lighttpd/1.4.35

OK
inadyn[1]: Successful alias table update for MY_DOMAIN_NAME.freemyip.com => new IP# MY_IP_ADDRESS
inadyn[1]: Updating cache for MY_DOMAIN_NAME.freemyip.com
```

**Comment by troglobit** [comment]
Thanks for the report.  Can you tell if there's a problem with the latest release as well, v2.6 tag?  I suspect this is a regression only on the master branch due to a refactor of the SSL/TLS support.

**Comment by queeup** [comment]
There is no problem on v2.6 tag.

**Comment by troglobit** [comment]
Great, thank you for checking!  Confirms my theory.  Marking this as a blocker for the next release.

**Comment by troglobit** [comment]
Very likely fixed in 14b3ee2 while adressing the Cloudflare bug.
